og_ui/src/EventSubscriber/OgUiEventSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Drupal\og_ui\EventSubscriber;
+
+use Drupal\og\Event\PermissionEventInterface;
+use Drupal\og\OgRoleInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscribers for Organic Groups.
+ */
+class OgUiEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      PermissionEventInterface::EVENT_NAME => [['provideDefaultOgPermissions']],
+    ];
+  }
```

Comment:
So here is how an implementation of `hook_og_permission()` looks as an event subscriber. This class will not only contain this single subscriber, but it can contain more subscribers. The events you want to react to are listed here in `getSubscribedEvents()`. You can even have multiple subscribers for the same event, you just provide a list of methods that need to be executed when the event fires.

If you want to change the order when your code is executed, you can add a priority number to it. If this is omitted, the priority is 0 by default. If you want your code to run at the end because you want to alter some data from other modules, you can for example decrease the priority of a particular implementation to -10:

```
  return [
    PermissionEventInterface::EVENT_NAME => [['provideDefaultOgPermissions', -10]],
  ];
```